content/tutorial-escrow.md

```diff
@@ -19,6 +19,409 @@ When testing in [stand-alone mode](concept-stand-alone-mode.html), you can force
+
+```python
+from time import time
+ripple_offset = 946684800
+cancel_after = int(time()) + (24*60*60) - 946684800
+print(cancel_after)
+# Example: 556927412
+```
+
+#}-->
+
+<!-- MULTICODE_BLOCK_END -->
+
+**Warning:** In the XRP Ledger, you must specify time as **seconds since the Ripple Epoch** (2000-01-01T00:00:00Z). If you use a UNIX time in the `CancelAfter` or `FinishAfter` field without converting to the equivalent Ripple time first, that sets the unlock time to an extra **30 years** in the future!
+
+### 3. Submit EscrowCreate transaction
+
+[Sign and submit](reference-transaction-format.html#signing-and-submitting-transactions) an [EscrowCreate transaction][]. Set the `Condition` field of the transaction to the time when the held payment should be released. Set the `Destination` to the recipient, which can be the same address as the sender. Include the `CancelAfter` or `FinishAfter` time you calculated in the previous step.
+
+{% include 'snippets/secret-key-warning.md' %}
+
+Request:
+
+<!-- MULTICODE_BLOCK_START -->
+
+*Websocket*
+
+```json
+{% include 'code_samples/escrow/websocket/submit-request-escrowcreate-condition.json' %}
+```
+
+<!-- MULTICODE_BLOCK_END -->
+
+Response:
+
+<!-- MULTICODE_BLOCK_START -->
+
+*Websocket*
+
+```json
+{% include 'code_samples/escrow/websocket/submit-response-escrowcreate-condition.json' %}
+```
+
+<!-- MULTICODE_BLOCK_END -->
+
+### 4. Wait for validation
+
+{% include 'snippets/wait-for-validation.md' %}
+
+### 5. Confirm that the escrow was created
+
+Use the [`tx` command](reference-rippled.html#tx) with the transaction's identifying hash to check its final status. In particular, look for a `CreatedNode` in the transaction metadata to indicate that it created an [Escrow ledger object](reference-ledger-format.html#escrow).
+
+Request:
+
+<!-- MULTICODE_BLOCK_START -->
+
+*Websocket*
+
+```json
+{% include 'code_samples/escrow/websocket/tx-request-escrowcreate-condition.json' %}
+```
+
+<!-- MULTICODE_BLOCK_END -->
+
+Response:
+
+<!-- MULTICODE_BLOCK_START -->
+
+*Websocket*
+
+```json
+{% include 'code_samples/escrow/websocket/tx-response-escrowcreate-condition.json' %}
+```
+
+<!-- MULTICODE_BLOCK_END -->
+
+### 6. Submit EscrowFinish transaction
+
+[Sign and submit](reference-transaction-format.html#signing-and-submitting-transactions) an [EscrowCreate transaction][] to execute the release of the funds after the `FinishAfter` time has passed. Set the `Owner` field of the transaction to the `Account` address from the EscrowCreate transaction, and the `OfferSequence` to the `Sequence` number from the EscrowCreate transaction. Set the `Condition` and `Fulfillment` fields to the condition and fulfillment values, in hexadecimal, that you generated in step 1. Set the `Fee` ([transaction cost](concept-transaction-cost.html)) value based on the size of the fulfillment in bytes: a conditional EscrowFinish requires at least 330 drops of XRP plus 10 drops per 16 bytes in the size of the fulfillment.
```

Comment:
1) Yes, it should be EscrowFinish; good catch.
2) The term of "base fee" is used inconsistently in the `rippled` source, but 330 drops + 10 drops per 16 bytes is a simplification of the actual formula on the assumption that the reference fee for a transaction is 10 drops. The [real formula](https://github.com/ripple/rippled/blob/develop/src/ripple/app/tx/impl/Escrow.cpp#L345-L355) for an EscrowFinish with a Fulfillment is:

        GenericFee + (BaseFee * (32 + (FulfillmentSize / 16)))

    What I'm calling `GenericFee` is the result of the generic [`Transactor::calculateBaseFee` calculation](https://github.com/ripple/rippled/blob/develop/src/ripple/app/tx/impl/Transactor.cpp#L137-L154), which is the cost of the transaction after adjusting for multi-signatures. That formula is itself `BaseFee + (signerCount * BaseFee)`

    I guess one could simplify the overall formula for an EscrowFinish with Fulfillment to be as follows:

        BaseFee * (SignerCount + 33 + (FulfillmentSize / 16))

    (It's 33 because the GenericFee includes at minimum one multiple of the BaseFee.)

Reply:
Thanks.  I'll leave to your discretion whether you want to a) exclude the extra detail b) include the simplified formula, or c) put in a link to a thorough explanation.  I understand that there's a tension between completeness and readability.